modules/common/affinity/types.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2024 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +kubebuilder:object:generate:=true
+
+package affinity
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// OverrideSpec -
+type OverrideSpec struct {
```

Comment:
What's the purpose of creating this `OverrideSpec` type when it's (nearly) identical to corev1 `Affinity` structure, https://pkg.go.dev/k8s.io/api/core/v1#Affinity?

Reply:
I used it to distinguish in between the Affinity struct that is present in corev1 and the Override we build through the Topology [1] and we pass to:

```golang
func toCoreAffinity(
	affinity *v1.Affinity,
	override *OverrideSpec,
```
It simply helped to make the reading easier and avoid creating aliasing w/ the `corev1` struct. If it helps to reduce the amount of code involved in this patch, I can experiment to rely on the existing struct and see how it goes.

Reply:
I think the 'override' parameter name is sufficient to distinguish the parameters. However, I am (partially) persuaded by your argument that having our own structure definition makes it easier to modify the code if our structure needs to diverge from the v1.Affinity structure.

Thinking about it some more "OverrideSpec" is quite a generic name. How do you feel about  "AffinityOverride" for the name?

Also, as long as the two structures are essentially the same (no divergence yet) you could 'inline' the v1.Affinity structure and not have to essentially copy it.

Reply:
while they are identical then can it just be `type OverrideSpec corev1.Affinity`

Reply:
even better!

Reply:
> I think the 'override' parameter name is sufficient to distinguish the parameters. However, I am (partially) persuaded by your argument that having our own structure definition makes it easier to modify the code if our structure needs to diverge from the v1.Affinity structure.
> 
> Thinking about it some more "OverrideSpec" is quite a generic name. How do you feel about "AffinityOverride" for the name?
> 
Originally it was `AffinityOverride`, but `golint` complains when you import it elsewhere because it would be `affinity.AffinityOverride`. For this reason I turned it into `OverrideSpec`.

> Also, as long as the two structures are essentially the same (no divergence yet) you could 'inline' the v1.Affinity structure and not have to essentially copy it.

Reply:
> while they are identical then can it just be `type OverrideSpec corev1.Affinity`

If we agree to rely on `corev1.Affinity` I'm inclined to remove `OverrideSpec` struct entirely. Are we in agreement with this?

Reply:
Ollie's suggestion looks like a reasonable compromise. It leverages the existing v1.Affinity (no need to repeat the details in our own structure) but establishes our own type name in case we need to diverge in the future.

And argh, I hate that tools like golint downvote using names that make better sense to us humans.